### PR TITLE
[Koin Project][Chore] DeptRepository UseCase 단위 테스트 추가

### PR DESCRIPTION
**Pull request by kongwoojin**
## PR 개요
이슈 번호: #1288

## PR 체크리스트
- [x] Code convention을 잘 지켰나요?
- [x] Lint check를 수행하였나요?
- [x] Assignees를 추가했나요?

## 작업사항
- [ ] 버그 수정
- [ ] 신규 기능
- [ ] 코드 스타일 수정 (포맷팅 등)
- [ ] 리팩토링 (기능 수정 X, API 수정 X)
- [x] 기타

### 작업사항의 상세한 설명
DeptRepository 관련 UseCase 단위 테스트 추가

- `FakeDeptRepository` 추가
- `FakeDeptErrorHandler` 추가
- `GetDeptNameFromStudentIdUseCaseTest` 추가 (8개 테스트)
- `GetDeptNamesUseCaseTest` 추가 (3개 테스트)

## 추가내용
- [x] develop, sprint 브랜치를 향하고 있습니다
- [ ] production 브랜치를 향하고 있습니다

## Summary by CodeRabbit

* **Tests**
  * Added test helpers that provide in-memory department data and deterministic error responses with configurable failure modes for repository behavior.
  * Added comprehensive unit tests for department-related logic covering valid student IDs, edge cases, empty results, non-existent departments, and repository failure scenarios.

### CODE REVIEW

File: domain/src/test/java/in/koreatech/koin/domain/usecase/dept/GetDeptNameFromStudentIdUseCaseTest.kt [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+
+    @Before
+    fun setUp() {
+        deptRepository = FakeDeptRepository()
+        deptErrorHandler = FakeDeptErrorHandler()
+    }
+
+    @Test
+    fun `유효한 10자리 학번으로 학과명을 가져온다`() = runTest {
+        deptRepository.setDeptNameForCode(deptCode10, "컴퓨터공학부")
```
Comment (KYM-P) [inline domain/src/test/java/in/koreatech/koin/domain/usecase/dept/GetDeptNameFromStudentIdUseCaseTest.kt:37]: 인(?)적 오류의 예방으로
학과 또한 const 를 쓰는건 어떨려나요

**Review by KYM-P — APPROVED** [review]
GOOD
